### ISSUE TITLE
The priority field does not take effect, and there is no priority scheduling

### ISSUE THREAD

**Issue by sanqiuli** [argoproj/argo-workflows#10564]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

I want to test how the pod is scheduled when the cluster resources are insufficient. My current cluster only has 3 cores left, and only 3 pods can run at the same time. For this purpose, the priority field is set to 0/20000/99999. When the priority value is large The pending pods of the workflow are not prioritized for scheduling. It is observed that the scheduling of the pods is out of order. I would like to know how the priority field schedules the pending pods?
![image](https://user-images.githubusercontent.com/61610792/220505665-711d15aa-2991-4f44-9f67-90da9302b6a4.png)

### Version

v3.4.1

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: loops-
spec:
  entrypoint: loop-example
  priority: 9000
  templates:
  - name: loop-example
    steps:
    - - name: print-message
        template: whalesay
        arguments:
          parameters:
          - name: message
            value: "{{item}}"
        withItems:
        - hello world
        - goodbye world
        - w
        - o
        - l
        - d
        - h
        - e
        - l
        - l
        - o

  - name: whalesay
    inputs:
      parameters:
      - name: message
    podSpecPatch: '{"containers":[{"name":"main","resources":{"requests":{"cpu":"1","memory":"1Gi"}}}]}'
    container:
      image: docker/whalesay:latest
      command: ["sh","-c"]
      args: ["sleep 30s"]
```

### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
time="2023-02-22T02:37:52.512Z" level=info msg="node changed" namespace=argo new.message=PodInitializing new.phase=Pending new.progress=0/1 nodeID=loops-wtwdf-83582783 old.message="Unschedulable: 0/4 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 3 Insufficient cpu." old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:37:52.512Z" level=info msg="node changed" namespace=argo new.message= new.phase=Succeeded new.progress=0/1 nodeID=loops-wtwdf-2425056532 old.message= old.phase=Running old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:37:52.512Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1472822349 workflow=loops-wtwdf
time="2023-02-22T02:37:52.514Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:37:52.514Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:37:52.514Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:37:52.522Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638007 workflow=loops-wtwdf
time="2023-02-22T02:37:52.528Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo/loops-wtwdf-whalesay-202212101/labelPodCompleted
time="2023-02-22T02:37:52.528Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo/loops-wtwdf-whalesay-2425056532/labelPodCompleted
time="2023-02-22T02:38:02.537Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:02.537Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:38:02.537Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1472822349 workflow=loops-wtwdf
time="2023-02-22T02:38:02.537Z" level=info msg="node changed" namespace=argo new.message= new.phase=Running new.progress=0/1 nodeID=loops-wtwdf-767782863 old.message=PodInitializing old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:38:02.537Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:38:02.538Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-3183890160 workflow=loops-wtwdf
time="2023-02-22T02:38:02.538Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-83582783 workflow=loops-wtwdf
time="2023-02-22T02:38:02.538Z" level=info msg="node changed" namespace=argo new.message=PodInitializing new.phase=Pending new.progress=0/1 nodeID=loops-wtwdf-1672472937 old.message="Unschedulable: 0/4 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 3 Insufficient cpu." old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:38:02.539Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:02.539Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:02.539Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:02.548Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638062 workflow=loops-wtwdf
time="2023-02-22T02:38:12.524Z" level=info msg="cleaning up pod" action=killContainers key=argo/loops-wtwdf-whalesay-202212101/killContainers
time="2023-02-22T02:38:39.866Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:39.866Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:38:39.866Z" level=info msg="node changed" namespace=argo new.message=PodInitializing new.phase=Pending new.progress=0/1 nodeID=loops-wtwdf-3183890160 old.message="Unschedulable: 0/4 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 3 Insufficient cpu." old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:38:39.866Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1672472937 workflow=loops-wtwdf
time="2023-02-22T02:38:39.866Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-83582783 workflow=loops-wtwdf
time="2023-02-22T02:38:39.866Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1472822349 workflow=loops-wtwdf
time="2023-02-22T02:38:39.867Z" level=info msg="node changed" namespace=argo new.message= new.phase=Succeeded new.progress=0/1 nodeID=loops-wtwdf-767782863 old.message= old.phase=Running old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:38:39.867Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:38:39.868Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:39.868Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:39.868Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:39.876Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638200 workflow=loops-wtwdf
time="2023-02-22T02:38:39.882Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo/loops-wtwdf-whalesay-767782863/labelPodCompleted
time="2023-02-22T02:38:49.891Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:49.891Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:38:49.891Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:38:49.891Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-3183890160 workflow=loops-wtwdf
time="2023-02-22T02:38:49.891Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1672472937 workflow=loops-wtwdf
time="2023-02-22T02:38:49.891Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1472822349 workflow=loops-wtwdf
time="2023-02-22T02:38:49.891Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-83582783 workflow=loops-wtwdf
time="2023-02-22T02:38:49.892Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:49.892Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:38:49.892Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:06.933Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:06.933Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:39:06.933Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1472822349 workflow=loops-wtwdf
time="2023-02-22T02:39:06.933Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1672472937 workflow=loops-wtwdf
time="2023-02-22T02:39:06.933Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-3183890160 workflow=loops-wtwdf
time="2023-02-22T02:39:06.933Z" level=info msg="node changed" namespace=argo new.message= new.phase=Running new.progress=0/1 nodeID=loops-wtwdf-83582783 old.message=PodInitializing old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:06.933Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:39:06.935Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:06.935Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:06.935Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:06.941Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638293 workflow=loops-wtwdf
time="2023-02-22T02:39:19.335Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:19.335Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:39:19.335Z" level=info msg="node changed" namespace=argo new.message= new.phase=Running new.progress=0/1 nodeID=loops-wtwdf-3183890160 old.message=PodInitializing old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:19.335Z" level=info msg="node changed" namespace=argo new.message= new.phase=Running new.progress=0/1 nodeID=loops-wtwdf-1672472937 old.message=PodInitializing old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:19.335Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1472822349 workflow=loops-wtwdf
time="2023-02-22T02:39:19.335Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-83582783 workflow=loops-wtwdf
time="2023-02-22T02:39:19.335Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:39:19.336Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:19.336Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:19.336Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:19.347Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638338 workflow=loops-wtwdf
time="2023-02-22T02:39:37.683Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:37.683Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:39:37.683Z" level=info msg="node changed" namespace=argo new.message=PodInitializing new.phase=Pending new.progress=0/1 nodeID=loops-wtwdf-1472822349 old.message="Unschedulable: 0/4 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 3 Insufficient cpu." old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:37.683Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-3183890160 workflow=loops-wtwdf
time="2023-02-22T02:39:37.683Z" level=info msg="node changed" namespace=argo new.message= new.phase=Succeeded new.progress=0/1 nodeID=loops-wtwdf-83582783 old.message= old.phase=Running old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:37.683Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:39:37.683Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1672472937 workflow=loops-wtwdf
time="2023-02-22T02:39:37.684Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:37.684Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:37.684Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:37.692Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638408 workflow=loops-wtwdf
time="2023-02-22T02:39:37.698Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo/loops-wtwdf-whalesay-83582783/labelPodCompleted
time="2023-02-22T02:39:47.707Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:47.707Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:39:47.707Z" level=info msg="node changed" namespace=argo new.message= new.phase=Running new.progress=0/1 nodeID=loops-wtwdf-1472822349 old.message=PodInitializing old.phase=Pending old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:47.707Z" level=info msg="node changed" namespace=argo new.message= new.phase=Succeeded new.progress=0/1 nodeID=loops-wtwdf-3183890160 old.message= old.phase=Running old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:47.707Z" level=info msg="node changed" namespace=argo new.message= new.phase=Succeeded new.progress=0/1 nodeID=loops-wtwdf-1672472937 old.message= old.phase=Running old.progress=0/1 workflow=loops-wtwdf
time="2023-02-22T02:39:47.707Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:39:47.708Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:47.708Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:47.708Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:47.717Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638460 workflow=loops-wtwdf
time="2023-02-22T02:39:47.723Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo/loops-wtwdf-whalesay-3183890160/labelPodCompleted
time="2023-02-22T02:39:47.723Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo/loops-wtwdf-whalesay-1672472937/labelPodCompleted
time="2023-02-22T02:39:52.407Z" level=info msg="Processing workflow" namespace=argo workflow=loops-jz86m
time="2023-02-22T02:39:52.407Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-jz86m
time="2023-02-22T02:39:52.407Z" level=info msg="node unchanged" namespace=argo nodeID=loops-jz86m-2228216424 workflow=loops-jz86m
time="2023-02-22T02:39:52.407Z" level=info msg="node changed" namespace=argo new.message= new.phase=Running new.progress=0/1 nodeID=loops-jz86m-3298615862 old.message="Unschedulable: 0/4 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 3 Insufficient cpu." old.phase=Pending old.progress=0/1 workflow=loops-jz86m
time="2023-02-22T02:39:52.407Z" level=info msg="node unchanged" namespace=argo nodeID=loops-jz86m-1451398583 workflow=loops-jz86m
time="2023-02-22T02:39:52.407Z" level=info msg="node unchanged" namespace=argo nodeID=loops-jz86m-3435209314 workflow=loops-jz86m
time="2023-02-22T02:39:52.409Z" level=info msg="Workflow step group node loops-jz86m-162475158 not yet completed" namespace=argo workflow=loops-jz86m
time="2023-02-22T02:39:52.409Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-jz86m
time="2023-02-22T02:39:52.409Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-jz86m
time="2023-02-22T02:39:52.417Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638481 workflow=loops-jz86m
time="2023-02-22T02:39:57.718Z" level=info msg="Processing workflow" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:57.719Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-wtwdf
time="2023-02-22T02:39:57.719Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1488395053 workflow=loops-wtwdf
time="2023-02-22T02:39:57.719Z" level=info msg="node unchanged" namespace=argo nodeID=loops-wtwdf-1472822349 workflow=loops-wtwdf
time="2023-02-22T02:39:57.720Z" level=info msg="Workflow step group node loops-wtwdf-947465999 not yet completed" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:57.720Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:39:57.720Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-wtwdf
time="2023-02-22T02:40:02.418Z" level=info msg="Processing workflow" namespace=argo workflow=loops-jz86m
time="2023-02-22T02:40:02.419Z" level=info msg="Task-result reconciliation" namespace=argo numObjs=0 workflow=loops-jz86m
time="2023-02-22T02:40:02.419Z" level=info msg="node unchanged" namespace=argo nodeID=loops-jz86m-2228216424 workflow=loops-jz86m
time="2023-02-22T02:40:02.419Z" level=info msg="node changed" namespace=argo new.message= new.phase=Running new.progress=0/1 nodeID=loops-jz86m-1451398583 old.message="Unschedulable: 0/4 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate, 3 Insufficient cpu." old.phase=Pending old.progress=0/1 workflow=loops-jz86m
time="2023-02-22T02:40:02.419Z" level=info msg="node unchanged" namespace=argo nodeID=loops-jz86m-3435209314 workflow=loops-jz86m
time="2023-02-22T02:40:02.419Z" level=info msg="node unchanged" namespace=argo nodeID=loops-jz86m-3298615862 workflow=loops-jz86m
time="2023-02-22T02:40:02.420Z" level=info msg="Workflow step group node loops-jz86m-162475158 not yet completed" namespace=argo workflow=loops-jz86m
time="2023-02-22T02:40:02.420Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=loops-jz86m
time="2023-02-22T02:40:02.420Z" level=info msg=reconcileAgentPod namespace=argo workflow=loops-jz86m
time="2023-02-22T02:40:02.430Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=39638526 workflow=loops-jz86m
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
time="2023-02-22T02:40:14.950Z" level=info msg="Deadline monitor stopped"
time="2023-02-22T02:40:14.951Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-02-22T02:40:14.951Z" level=info msg="Alloc=6213 TotalAlloc=12119 Sys=23762 NumGC=4 Goroutines=4"
time="2023-02-22T02:34:02.516Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-02-22T02:34:02.516Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=argo podName=loops-wtwdf-whalesay-4203342452 template="{\"name\":\"whalesay\",\"inputs\":{\"parameters\":[{\"name\":\"message\",\"value\":\"w\"}]},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"command\":[\"sh\",\"-c\"],\"args\":[\"sleep 30s\"],\"resources\":{}},\"podSpecPatch\":\"{\\\"containers\\\":[{\\\"name\\\":\\\"main\\\",\\\"resources\\\":{\\\"requests\\\":{\\\"cpu\\\":\\\"1\\\",\\\"memory\\\":\\\"1Gi\\\"}}}]}\"}" version="&Version{Version:v3.4.1,BuildDate:2022-10-01T15:03:42Z,GitCommit:0546fef0b096d84c9e3362d2b241614e743ebe97,GitTag:v3.4.1,GitTreeState:clean,GoVersion:go1.18.6,Compiler:gc,Platform:linux/amd64,}"
time="2023-02-22T02:34:02.516Z" level=info msg="Starting deadline monitor"
time="2023-02-22T02:35:57.623Z" level=info msg="Main container completed" error="<nil>"
time="2023-02-22T02:35:57.646Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2023-02-22T02:35:57.646Z" level=info msg="No output parameters"
time="2023-02-22T02:35:57.646Z" level=info msg="No output artifacts"
time="2023-02-22T02:35:57.646Z" level=info msg="Deadline monitor stopped"
time="2023-02-22T02:35:57.646Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-02-22T02:35:57.664Z" level=info msg="Alloc=8030 TotalAlloc=12288 Sys=18642 NumGC=3 Goroutines=4"
time="2023-02-22T02:36:01.306Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-02-22T02:36:01.306Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=argo podName=loops-wtwdf-whalesay-602191333 template="{\"name\":\"whalesay\",\"inputs\":{\"parameters\":[{\"name\":\"message\",\"value\":\"o\"}]},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"command\":[\"sh\",\"-c\"],\"args\":[\"sleep 30s\"],\"resources\":{}},\"podSpecPatch\":\"{\\\"containers\\\":[{\\\"name\\\":\\\"main\\\",\\\"resources\\\":{\\\"requests\\\":{\\\"cpu\\\":\\\"1\\\",\\\"memory\\\":\\\"1Gi\\\"}}}]}\"}" version="&Version{Version:v3.4.1,BuildDate:2022-10-01T15:03:42Z,GitCommit:0546fef0b096d84c9e3362d2b241614e743ebe97,GitTag:v3.4.1,GitTreeState:clean,GoVersion:go1.18.6,Compiler:gc,Platform:linux/amd64,}"
time="2023-02-22T02:36:01.306Z" level=info msg="Starting deadline monitor"
time="2023-02-22T02:37:24.346Z" level=info msg="Main container completed" error="<nil>"
time="2023-02-22T02:37:24.346Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2023-02-22T02:37:24.346Z" level=info msg="No output parameters"
time="2023-02-22T02:37:24.346Z" level=info msg="No output artifacts"
time="2023-02-22T02:37:24.346Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-02-22T02:37:24.346Z" level=info msg="Deadline monitor stopped"
time="2023-02-22T02:37:24.347Z" level=info msg="Alloc=6891 TotalAlloc=12111 Sys=18898 NumGC=3 Goroutines=5"
time="2023-02-22T02:38:35.244Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-02-22T02:38:35.245Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=argo podName=loops-wtwdf-whalesay-3183890160 template="{\"name\":\"whalesay\",\"inputs\":{\"parameters\":[{\"name\":\"message\",\"value\":\"l\"}]},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"command\":[\"sh\",\"-c\"],\"args\":[\"sleep 30s\"],\"resources\":{}},\"podSpecPatch\":\"{\\\"containers\\\":[{\\\"name\\\":\\\"main\\\",\\\"resources\\\":{\\\"requests\\\":{\\\"cpu\\\":\\\"1\\\",\\\"memory\\\":\\\"1Gi\\\"}}}]}\"}" version="&Version{Version:v3.4.1,BuildDate:2022-10-01T15:03:42Z,GitCommit:0546fef0b096d84c9e3362d2b241614e743ebe97,GitTag:v3.4.1,GitTreeState:clean,GoVersion:go1.18.6,Compiler:gc,Platform:linux/amd64,}"
time="2023-02-22T02:38:35.245Z" level=info msg="Starting deadline monitor"
time="2023-02-22T02:39:39.278Z" level=info msg="Main container completed" error="<nil>"
time="2023-02-22T02:39:39.278Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2023-02-22T02:39:39.278Z" level=info msg="No output parameters"
time="2023-02-22T02:39:39.278Z" level=info msg="No output artifacts"
time="2023-02-22T02:39:39.279Z" level=info msg="Deadline monitor stopped"
time="2023-02-22T02:39:39.279Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-02-22T02:39:39.279Z" level=info msg="Alloc=7619 TotalAlloc=12178 Sys=18898 NumGC=3 Goroutines=4"
time="2023-02-22T02:40:41.646Z" level=info msg="Starting Workflow Executor" version=v3.4.1
time="2023-02-22T02:40:41.651Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-02-22T02:40:41.651Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=argo podName=loops-wtwdf-whalesay-1488395053 template="{\"name\":\"whalesay\",\"inputs\":{\"parameters\":[{\"name\":\"message\",\"value\":\"goodbye world\"}]},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"command\":[\"sh\",\"-c\"],\"args\":[\"sleep 30s\"],\"resources\":{}},\"podSpecPatch\":\"{\\\"containers\\\":[{\\\"name\\\":\\\"main\\\",\\\"resources\\\":{\\\"requests\\\":{\\\"cpu\\\":\\\"1\\\",\\\"memory\\\":\\\"1Gi\\\"}}}]}\"}" version="&Version{Version:v3.4.1,BuildDate:2022-10-01T15:03:42Z,GitCommit:0546fef0b096d84c9e3362d2b241614e743ebe97,GitTag:v3.4.1,GitTreeState:clean,GoVersion:go1.18.6,Compiler:gc,Platform:linux/amd64,}"
time="2023-02-22T02:40:41.652Z" level=info msg="Starting deadline monitor"
time="2023-02-22T02:37:59.146Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-02-22T02:37:59.146Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=argo podName=loops-wtwdf-whalesay-1672472937 template="{\"name\":\"whalesay\",\"inputs\":{\"parameters\":[{\"name\":\"message\",\"value\":\"l\"}]},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"command\":[\"sh\",\"-c\"],\"args\":[\"sleep 30s\"],\"resources\":{}},\"podSpecPatch\":\"{\\\"containers\\\":[{\\\"name\\\":\\\"main\\\",\\\"resources\\\":{\\\"requests\\\":{\\\"cpu\\\":\\\"1\\\",\\\"memory\\\":\\\"1Gi\\\"}}}]}\"}" version="&Version{Version:v3.4.1,BuildDate:2022-10-01T15:03:42Z,GitCommit:0546fef0b096d84c9e3362d2b241614e743ebe97,GitTag:v3.4.1,GitTreeState:clean,GoVersion:go1.18.6,Compiler:gc,Platform:linux/amd64,}"
time="2023-02-22T02:37:59.146Z" level=info msg="Starting deadline monitor"
time="2023-02-22T02:39:41.173Z" level=info msg="Main container completed" error="<nil>"
time="2023-02-22T02:39:41.173Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2023-02-22T02:39:41.173Z" level=info msg="No output parameters"
time="2023-02-22T02:39:41.173Z" level=info msg="No output artifacts"
time="2023-02-22T02:39:41.173Z" level=info msg="Deadline monitor stopped"
time="2023-02-22T02:39:41.173Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-02-22T02:39:41.174Z" level=info msg="Alloc=7055 TotalAlloc=12269 Sys=24274 NumGC=4 Goroutines=4"
time="2023-02-22T02:36:11.177Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-02-22T02:36:11.177Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=argo podName=loops-wtwdf-whalesay-202212101 template="{\"name\":\"whalesay\",\"inputs\":{\"parameters\":[{\"name\":\"message\",\"value\":\"l\"}]},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"command\":[\"sh\",\"-c\"],\"args\":[\"sleep 30s\"],\"resources\":{}},\"podSpecPatch\":\"{\\\"containers\\\":[{\\\"name\\\":\\\"main\\\",\\\"resources\\\":{\\\"requests\\\":{\\\"cpu\\\":\\\"1\\\",\\\"memory\\\":\\\"1Gi\\\"}}}]}\"}" version="&Version{Version:v3.4.1,BuildDate:2022-10-01T15:03:42Z,GitCommit:0546fef0b096d84c9e3362d2b241614e743ebe97,GitTag:v3.4.1,GitTreeState:clean,GoVersion:go1.18.6,Compiler:gc,Platform:linux/amd64,}"
time="2023-02-22T02:36:11.177Z" level=info msg="Starting deadline monitor"
time="2023-02-22T02:37:42.213Z" level=info msg="Main container completed" error="<nil>"
time="2023-02-22T02:37:42.213Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2023-02-22T02:37:42.213Z" level=info msg="No output parameters"
time="2023-02-22T02:37:42.213Z" level=info msg="No output artifacts"
time="2023-02-22T02:37:42.213Z" level=info msg="Deadline monitor stopped"
time="2023-02-22T02:37:42.213Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-02-22T02:37:42.214Z" level=info msg="Alloc=6226 TotalAlloc=12242 Sys=19410 NumGC=4 Goroutines=4"
time="2023-02-22T02:36:11.853Z" level=info msg="Using executor retry strategy" Duration=1s Factor=1.6 Jitter=0.5 Steps=5
time="2023-02-22T02:36:11.853Z" level=info msg="Executor initialized" deadline="0001-01-01 00:00:00 +0000 UTC" includeScriptOutput=false namespace=argo podName=loops-wtwdf-whalesay-2425056532 template="{\"name\":\"whalesay\",\"inputs\":{\"parameters\":[{\"name\":\"message\",\"value\":\"d\"}]},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"command\":[\"sh\",\"-c\"],\"args\":[\"sleep 30s\"],\"resources\":{}},\"podSpecPatch\":\"{\\\"containers\\\":[{\\\"name\\\":\\\"main\\\",\\\"resources\\\":{\\\"requests\\\":{\\\"cpu\\\":\\\"1\\\",\\\"memory\\\":\\\"1Gi\\\"}}}]}\"}" version="&Version{Version:v3.4.1,BuildDate:2022-10-01T15:03:42Z,GitCommit:0546fef0b096d84c9e3362d2b241614e743ebe97,GitTag:v3.4.1,GitTreeState:clean,GoVersion:go1.18.6,Compiler:gc,Platform:linux/amd64,}"
time="2023-02-22T02:36:11.853Z" level=info msg="Starting deadline monitor"
time="2023-02-22T02:37:44.881Z" level=info msg="Main container completed" error="<nil>"
time="2023-02-22T02:37:44.881Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2023-02-22T02:37:44.944Z" level=info msg="No output parameters"
time="2023-02-22T02:37:44.944Z" level=info msg="No output artifacts"
time="2023-02-22T02:37:44.944Z" level=info msg="stopping progress monitor (context done)" error="context canceled"
time="2023-02-22T02:37:44.944Z" level=info msg="Deadline monitor stopped"
time="2023-02-22T02:37:45.215Z" level=info msg="Alloc=6923 TotalAlloc=12234 Sys=20178 NumGC=4 Goroutines=6"
```

**Comment by tico24** [comment]
I agree, it doesn't seem to work. I wonder if it's because it's based on the old kubernetes pod priority. As a workaround, you could try switching to using PriorityClassNames, and defining appropriate priorityclasses? https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass

**Comment by sanqiuli** [comment]
> PriorityClassName

In the v3.4.1 version, the priorityClassName field is invalid, argo submit will appear `Failed to parse workflow: json: unknown field "priorityClassName"` , I replaced the` priorityClassName` with a `podPriorityClassName`, this is valid

**Comment by juliev0** [comment]
Do you mean that if there are already n Pods running of lower priority and parallelism=n, then a new Pod won't cause another to be evicted?

**Comment by JPZ13** [comment]
@tico24 - I'm in the weekly bug triage meeting, and we're looking for a bit more color on this issue. Could you record a quick demo video of what you're seeing and describe the expected behavior and post it on this thread?

**Comment by JPZ13** [comment]
@sanqiuli - are you unblocked with the name switch to `podPriorityClassName`? If so, can we close the issue?

**Comment by sanqiuli** [comment]
> @sanqiuli - are you unblocked with the name switch to `podPriorityClassName`? If so, can we close the issue?

yes,it is.`podPriorityClassName` it works
